### PR TITLE
Fix: backward report build

### PR DESCRIPTION
**Pull request by F4ever**

### CODE REVIEW

File: src/providers/execution/contracts/staking_router.py [hunk 1]
```diff
@@ -58,3 +58,30 @@ def get_all_node_operator_digests(self, module: StakingModule, block_identifier:
 
 class StakingRouterContractV2(StakingRouterContract):
     abi_path = './assets/StakingRouterV2.json'
+
+    @lru_cache(maxsize=1)
```
Comment (vgorkavenko) [inline src/providers/execution/contracts/staking_router.py:62]: not sure that it is good idea to have cache and `block_identifier = 'latest'` by default at the same time

Comment (F4ever) [reply src/providers/execution/contracts/staking_router.py:62]: This is a special cache, that ignores 'latest' and default values and caches only hashes and numbers

File: src/providers/execution/contracts/staking_router.py [hunk 2]
```diff
@@ -58,3 +58,30 @@ def get_all_node_operator_digests(self, module: StakingModule, block_identifier:
...
+    def get_staking_modules(self, block_identifier: BlockIdentifier = 'latest') -> list[StakingModule]:
+        """
+        Returns all registered staking modules
+        """
+        contract_version = self.get_contract_version(block_identifier)
+
+        if contract_version == 1:
+            # Backward compatibility in case if new oracle have to build report for old protocol version
+            # But latest contracts has new version
+            staking_router = self.w3.eth.contract(
```
Comment (vgorkavenko) [inline src/providers/execution/contracts/staking_router.py:73]: let's log that we call an old abi. here or instead of logging at the end of function

Comment (F4ever) [reply src/providers/execution/contracts/staking_router.py:74]: fixed

**Review by vgorkavenko — APPROVED** [review]

**Pull request merged**